go/api/config.go

```diff
@@ -0,0 +1,309 @@
+	}
+
+	return protobuf.ReadFrom_Primary
+}
+
+type baseClientConfiguration struct {
+	addresses      []NodeAddress
+	useTLS         bool
+	credentials    *RedisCredentials
+	readFrom       ReadFrom
+	requestTimeout int
+	clientName     string
+}
+
+func (config *baseClientConfiguration) toProtobuf() *protobuf.ConnectionRequest {
+	request := protobuf.ConnectionRequest{}
+	for _, address := range config.addresses {
+		request.Addresses = append(request.Addresses, address.toProtobuf())
+	}
+
+	if config.useTLS {
+		request.TlsMode = protobuf.TlsMode_SecureTls
+	} else {
+		request.TlsMode = protobuf.TlsMode_NoTls
+	}
+
+	if config.credentials != nil {
+		request.AuthenticationInfo = config.credentials.toProtobuf()
+	}
+
+	request.ReadFrom = mapReadFrom(config.readFrom)
+	if config.requestTimeout != 0 {
+		request.RequestTimeout = uint32(config.requestTimeout)
+	}
+
+	if config.clientName != "" {
+		request.ClientName = config.clientName
+	}
+
+	return &request
+}
+
+// BackoffStrategy represents the strategy used to determine how and when to reconnect, in case of connection failures. The
+// time between attempts grows exponentially, to the formula:
+//
+//	rand(0 ... factor * (exponentBase ^ N))
+//
+// where N is the number of failed attempts.
+//
+// Once the maximum value is reached, that will remain the time between retry attempts until a reconnect attempt is successful.
+// The client will attempt to reconnect indefinitely.
+type BackoffStrategy struct {
+	// Required: number of retry attempts that the client should perform when disconnected from the server, where the time
+	// between retries increases. Once the retries have reached the maximum value, the time between retries will remain
+	// constant until a reconnect attempt is successful.
+	numOfRetries int
+	// Required: the multiplier that will be applied to the waiting time between each retry.
+	factor int
+	// Required: the exponent base configured for the strategy.
+	exponentBase int
+}
+
+// NewBackoffStrategy returns a new BackoffStrategy with the given configuration parameters.
+func NewBackoffStrategy(numOfRetries int, factor int, exponentBase int) *BackoffStrategy {
+	return &BackoffStrategy{numOfRetries, factor, exponentBase}
+}
+
+func (strategy *BackoffStrategy) toProtobuf() *protobuf.ConnectionRetryStrategy {
+	return &protobuf.ConnectionRetryStrategy{
+		NumberOfRetries: uint32(strategy.numOfRetries),
+		Factor:          uint32(strategy.factor),
+		ExponentBase:    uint32(strategy.exponentBase),
+	}
+}
+
+// RedisClientConfiguration represents the configuration settings for a Standalone Redis client. baseClientConfiguration is an
+// embedded struct that contains shared settings for standalone and cluster clients.
+type RedisClientConfiguration struct {
+	baseClientConfiguration
+	reconnectStrategy *BackoffStrategy
+	databaseId        int
+}
+
+// NewRedisClientConfiguration returns a [RedisClientConfiguration] with default configuration settings. For further
```

Comment:
you're inconsistently using [Foo] / new Foo between docs.

Reply:
Good point, I'll fix that, thanks.